File: Backend/internal/data-structure/dto/customer.go
package dto

import (
	"simple-erp-service/internal/data-structure/models"
	"time"
)

// ApiCustomer representa os dados de cliente para exibição
type ApiCustomer struct {
	ID             uint              `json:"id"`
	FirstName      string            `json:"first_name"`
	LastName       string            `json:"last_name"`
	PersonType     string            `json:"person_type"`
	DocumentNumber string            `json:"document_number"`
	CompanyName    string            `json:"company_name"`
	IsActive       bool              `json:"is_active"`
	CreatedAt      time.Time         `json:"created_at"`
	UpdatedAt      time.Time         `json:"updated_at"`
	Addresses      *[]models.Address `json:"addresses"`
	Contacts       *[]models.Contact `json:"contacts"`
}

// ApiCustomerDetail representa os dados detalhados de cliente para exibição
type ApiCustomerDetail struct {
	ID             uint   `json:"id"`
	FirstName      string `json:"first_name"`
	LastName       string `json:"last_name"`
	PersonType     string `json:"person_type"`
	DocumentNumber string `json:"document_number"`
	CompanyName    string `json:"company_name"`
	IsActive       bool   `json:"is_active"`

	Addresses *[]models.Address `json:"addresses"`
	Contacts  *[]models.Contact `json:"contacts"`

	CreatedAt time.Time `json:"created_at"`
	CreatedBy *ApiUser  `json:"created_by"`

	UpdatedBy *ApiUser  `json:"updated_by"`
	UpdatedAt time.Time `json:"updated_at"`

	// Sales        *[]ApiSale        `json:"sales"`
	// Transactions *[]ApiTransaction `json:"transactions"`
}

// CustomerListDTO representa uma lista paginada de clientes
type ApiCustomerListPaginated struct {
	Customer   []ApiCustomer `json:"data"`
	Pagination ApiPagination `json:"pagination"`
}

// ApiCustomerFromModel converte um Customer para ApiCustomer
func ApiCustomerFromModel(c models.Customer) ApiCustomer {
	return ApiCustomer{
		ID:             c.ID,
		FirstName:      c.FirstName,
		LastName:       c.LastName,
		PersonType:     c.PersonType,
		DocumentNumber: c.DocumentNumber,
		CompanyName:    c.CompanyName,
		IsActive:       c.IsActive,
		CreatedAt:      c.CreatedAt,
		UpdatedAt:      c.UpdatedAt,
	}
}

// ApiCustomerDetailFromModel converte um Customer para ApiCustomerDetail
func ApiCustomerDetailFromModel(c models.Customer) ApiCustomerDetail {
	return ApiCustomerDetail{
		ID:             c.ID,
		FirstName:      c.FirstName,
		LastName:       c.LastName,
		PersonType:     c.PersonType,
		DocumentNumber: c.DocumentNumber,
		CompanyName:    c.CompanyName,
		IsActive:       c.IsActive,
		CreatedAt:      c.CreatedAt,
		UpdatedAt:      c.UpdatedAt,
	}
}

File: Frontend/src/types/customer.ts
import type { ApiPagination } from "./common";
import type { Address, Contact } from "./address-contact";

// Espelha Backend/microservices/api.core/internal/domain/customer.go

export type PersonType = "F" | "J";

export interface ApiCustomer {
  id: number;
  first_name: string;
  last_name: string;
  person_type: PersonType;
  document_number: string;
  company_name: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  addresses?: Address[];
  contacts?: Contact[];
}

export interface CreateCustomerRequest {
  first_name: string;
  last_name?: string;
  person_type: PersonType;
  document_number: string;
  company_name?: string;
  is_active: boolean;
  notes?: string;
}

export interface UpdateCustomerRequest {
  first_name: string;
  last_name?: string;
  document_number: string;
  company_name?: string;
  is_active: boolean;
  notes?: string;
}

export interface CustomerFilterRequest {
  id?: number;
  first_name?: string;
  last_name?: string;
  person_type?: string;
  document_number?: string;
  company_name?: string;
  is_active?: boolean;
  page_no?: number;
  page_size?: number;
  order_by_column?: string;
  is_asc?: boolean;
}

export interface ApiCustomerListPaginated {
  data: ApiCustomer[];
  pagination: ApiPagination;
}
